includes/Shopping/Product.php

```diff
@@ -127,6 +127,35 @@ public function __construct( array $product = [] ) {
 		}
 	}
 
+	/**
+	 * Convert array to object properties.
+	 *
+	 * @since 1.26.0
+	 *
+	 * @param array<string, mixed> $product Array of product.
+	 *
+	 * @phpstan-param ProductData $product
+	 */
+	public function load_from_array( array $product ): void {
```

Comment:
We should just make this a static factory method that returns a new `Product` instance. Have been thinking the same about the `Story` class too for. More straightforward to just write this:

`$product = Product::from_array( [ ... ] )`

Reply:
Fixed: https://github.com/GoogleForCreators/web-stories-wp/pull/12366/commits/0045c9ba9049b7ccbc192e2938b6315a307062dd